src/main/java/com/redhat/devtools/intellij/lsp4ij/internal/CancellationSupport.java

```diff
@@ -32,20 +32,25 @@ public class CancellationSupport implements CancelChecker {
 	private boolean cancelled;
 
 	public CancellationSupport() {
-		this.futuresToCancel = new ArrayList<CompletableFuture<?>>();
+		this.futuresToCancel = new CopyOnWriteArrayList<>();
 		this.cancelled = false;
 	}
 
 	public <T> CompletableFuture<T> execute(CompletableFuture<T> future) {
-		this.futuresToCancel.add(future);
+		if (cancelled) {
+			future.cancel(true);
+			checkCanceled();

```

Comment:
After reading again the lsp opérations which catch ProcessCanceledException I think we should throw ProcessCanceledException instead of calling checkedCancelled or update opérations to catch CancellationException

Reply:
@fbricon once the comment will be fixed we can merge it